mailet/standard/src/test/java/org/apache/james/transport/mailets/LogMessageTest.java

```diff
@@ -0,0 +1,202 @@
+        verifyNoMoreInteractions(logger);
+    }
+
+    @Test
+    public void serviceShouldLogWhenExceptionOccured() throws Exception {
+        mailet.init(mailetConfig);
+
+        Mail mail = mock(Mail.class);
+        when(mail.getName())
+            .thenReturn("name");
+        MessagingException messagingException = new MessagingException("exception message");
+        when(mail.getMessage())
+            .thenThrow(messagingException);
+
+        mailet.service(mail);
+
+        verify(logger).info("Logging mail name");
+        verify(logger).error("Error logging message.");
+        verify(logger).error("exception message");
+        verifyNoMoreInteractions(logger);
+    }
+
+    @Test
+    public void serviceShouldSetTheMailStateWhenPassThroughIsFalse() throws Exception {
+        mailetConfig.setProperty("passThrough", "false");
+        mailet.init(mailetConfig);
+
+        FakeMail mail = createMail();
+        mailet.service(mail);
+
+        assertThat(mail.getState()).isEqualTo(Mail.GHOST);
+    }
+
+    @Test
+    public void serviceShouldNotLogHeadersWhenFalse() throws Exception {
+        mailetConfig.setProperty("headers", "false");
+        mailet.init(mailetConfig);
+
+        mailet.service(createMail());
+
+        verify(logger).info("Logging mail name");
+        verify(logger).info("This is a fake mail");
+        verifyNoMoreInteractions(logger);
+    }
+
+    @Test
+    public void serviceShouldNotLogBodyWhenFalse() throws Exception {
+        mailetConfig.setProperty("body", "false");
+        mailet.init(mailetConfig);
+
+        mailet.service(createMail());
+
+        verify(logger).info("Logging mail name");
+        verify(logger).info("\n" +
+                "Subject: subject\n" +
+                "Content-Type: text/plain\n");
+        verifyNoMoreInteractions(logger);
+    }
+
+    @Test
+    public void serviceShouldNotLogFullBodyWhenBodyMaxIsSet() throws Exception {
+        mailetConfig.setProperty("maxBody", "2");
+        mailet.init(mailetConfig);
+
+        mailet.service(createMail());
+
+        verify(logger).info("Logging mail name");
+        verify(logger).info("\n" +
+                "Subject: subject\n" +
+                "Content-Type: text/plain\n");
+        verify(logger).info("Th");
+        verifyNoMoreInteractions(logger);
+    }
+
+    @Test
+    public void serviceShouldLogAdditionalCommentWhenCommentIsSet() throws Exception {
+        mailetConfig.setProperty("comment", "comment");
+        mailet.init(mailetConfig);
+
+        mailet.service(createMail());
+
+        verify(logger).info("Logging mail name");
+        verify(logger).info("comment");
+        verify(logger).info("\n" +
+                "Subject: subject\n" +
+                "Content-Type: text/plain\n");
+        verify(logger).info("This is a fake mail");
+        verifyNoMoreInteractions(logger);
+    }
+
+    private FakeMail createMail() throws MessagingException, AddressException {
```

Comment:
I really don't like this method, you probably don't need all this for every test. It just makes it harder to follow what's really tested